packages/amplify-migration-e2e/src/migrationReadmeParser.ts

```diff
@@ -44,12 +54,24 @@ function extractCommands(readmeContent: string) {
  *  --stack-name my-auth-stack-name
  * ```
  */
-export function getCommandsFromReadme(readmeContent: string) {
+export function getStackRefactorCommandsFromReadme(readmeContent: string) {
   const step1Content = extractContent(readmeContent, '### STEP 1', '#### Rollback step');
   const step2Content = extractContent(readmeContent, '### STEP 2', '#### Rollback step');
   const step3Content = extractContent(readmeContent, '### STEP 3', '#### Rollback step');
   const step1Commands = extractCommands(step1Content);
   const step2commands = extractCommands(step2Content);
   const step3Commands = extractCommands(step3Content);
+  // Pop first command from step3Commands
+  step3Commands.shift();
```

Comment:
why do we need to do this?

Reply:
This is done to keep the `step3commands[]` consistent across rollback commands and regular step commands. The first command in the step3 stackRefactorCommands is the setting of `BUCKET_NAME` env var which is missing in the rollback `step3RollbackCommands`.

Reply:
Added better clarifying comments. [ca7bc0a](https://github.com/aws-amplify/amplify-cli/pull/14009/commits/ca7bc0ae718820fc7206b924af2bdd15510f54a7)